### PR TITLE
Add pre-call view and network test

### PR DESCRIPTION
**Pull request by jeffswartz**

### CODE REVIEW

File: views/room.ejs [hunk 1]
```diff
@@ -23,9 +23,13 @@
     <link rel="chrome-webstore-item" href="https://chrome.google.com/webstore/detail/<%=chromeExtensionId%>">
     <link rel="stylesheet" href="https://assets.tokbox.com/solutions/css/style.css">
     <link rel="stylesheet" href="/css/annotation.opentok.css">
+    <link rel="stylesheet" href="/css/checkboxes.css" />
 
     <script>
-      window.isWebRTCVersion=<%=isWebRTCVersion%>
+      window.isWebRTCVersion=<%=isWebRTCVersion%>;
+      window.testSessionId='<%=testSessionId%>';
```
Comment (maikthomas) [inline views/room.ejs]: 1) The other variables we pass to the view are specifically to modify the view. These are just here to be accessed from the JS files. The convention for passing these variables appears to be via `getRoomInfo` in the roomController.
2) to me testSessionId and testToken sound like test variables (for testing the code) I would consider renaming as preCallToken or similar
WDYT @jeffswartz ?

File: web/js/roomController.js [hunk 2]
```diff
@@ -552,40 +572,77 @@
     }
   };
 
-  function showUserNamePrompt(roomName) {
+  function showCallSettingsPrompt(roomName, username) {
```
Comment (maikthomas) [inline web/js/roomController.js]: Any chance we can move this out of the roomController?
Perhaps not but I feel like this file is just growing and growing

**Review by maikthomas — COMMENTED** [review]
I've left a couple comments, there might be more things I've missed, lots of code here.
I'm a little worried about how much these files are growing.

File: views/room.ejs (hunk 1 above)
Comment (jeffswartz) [reply views/room.ejs]: Right. The `getRoomInfo` endpoint needs to know the userName (if there is one) to add it to the token, and we don't necessarily know that at this point. I could add a new endpoint `getPrecallInfo`, but that doesn't seem any better than passing the data in via ejs.

File: web/js/roomController.js (hunk 2 above)
Comment (jeffswartz) [reply web/js/roomController.js]: good point.

**Comment by jeffswartz** [comment]
Going to direct the PR to the ui-changes-sept-2017 branch of the opentok fork.

**Pull request closed without merging**